src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1432,22 +1437,94 @@ private void LoadAndValidateNetModuleAttributes(ref CustomAttributesBag<CSharpAt
+            Debug.Assert(attributeMatches is QuickAttributes.AssemblyKeyFile
+                or QuickAttributes.AssemblyKeyName
+                or QuickAttributes.AssemblySignatureKey);
+
+            if (_compilation.Options.OutputKind.IsNetModule())
             {
-                LoadAndValidateNetModuleAttributes(ref _lazyNetModuleAttributesBag);
+                return null;
+            }
+
+            var discardedDiagnostics = BindingDiagnosticBag.Discarded;
+
+            ImmutableArray<string> netModuleNames;
+            ImmutableArray<CSharpAttributeData> attributesFromNetModules = GetNetModuleAttributes(out netModuleNames);
+
+            WellKnownAttributeData wellKnownData = null;
+
+            if (attributesFromNetModules.Any())
+            {
+                wellKnownData = limitedDecodeWellKnownAttributes(attributesFromNetModules, netModuleNames, discardedDiagnostics);
+            }
+
+            discardedDiagnostics.Free();
+
+            return (CommonAssemblyWellKnownAttributeData)wellKnownData;
+
+            // Similar to ValidateAttributeUsageAndDecodeWellKnownAttributes, but doesn't load assembly-level attributes from source
+            // and only decodes 3 specific attributes.
+            WellKnownAttributeData limitedDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> attributesFromNetModules,
+                ImmutableArray<string> netModuleNames, BindingDiagnosticBag discardedDiagnostics)
+            {
+                Debug.Assert(attributesFromNetModules.Any());
+                Debug.Assert(netModuleNames.Any());
+                Debug.Assert(attributesFromNetModules.Length == netModuleNames.Length);
+
+                int netModuleAttributesCount = attributesFromNetModules.Length;
+
+                HashSet<CSharpAttributeData> uniqueAttributes = null;
+                CommonAssemblyWellKnownAttributeData result = null;
+
+                // Attributes from the second added module should override attributes from the first added module, etc.
+                // We don't reach here when the attribute was found in source already.
+                for (int i = netModuleAttributesCount - 1; i >= 0; i--)
+                {
+                    CSharpAttributeData attribute = attributesFromNetModules[i];
+                    if (!attribute.HasErrors && ValidateAttributeUsageForNetModuleAttribute(attribute, netModuleNames[i], discardedDiagnostics, ref uniqueAttributes))
+                    {
+                        limitedDecodeWellKnownAttribute(attribute, ref result);
+                    }
+                }
+
+                WellKnownAttributeData.Seal(result);
+                return result;
+            }
+
+            // Similar to DecodeWellKnownAttribute but only handles 3 specific attributes and ignores diagnostics.
+            void limitedDecodeWellKnownAttribute(CSharpAttributeData attribute, ref CommonAssemblyWellKnownAttributeData result)
```

Comment:
:memo: Existing tests exercise the last two cases (AssemblyKeyFile and AssemblyKeyName) and they break if incorrectly returning a null or empty string instead of values from attributes.
Added a test case for the first case (see test `AssemblySignatureKeyOnNetModule`).